pkg/cloudprovider/aws/instance.go

```diff
@@ -254,51 +254,39 @@ func (p *InstanceProvider) getInstance(ctx context.Context, id string) (*ec2.Ins
 	return instance, nil
 }
 
-func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) *v1.Node {
+func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType, amiFamily *string) *v1.Node {
 	for _, instanceType := range instanceTypes {
 		if instanceType.Name() == aws.StringValue(instance.InstanceType) {
 			nodeName := strings.ToLower(aws.StringValue(instance.PrivateDnsName))
 			if injection.GetOptions(ctx).GetAWSNodeNameConvention() == options.ResourceName {
 				nodeName = aws.StringValue(instance.InstanceId)
 			}
 
-			resources := v1.ResourceList{}
-			for resourceName, quantity := range map[v1.ResourceName]resource.Quantity{
-				v1.ResourcePods:             instanceType.Resources()[v1.ResourcePods],
-				v1.ResourceCPU:              instanceType.Resources()[v1.ResourceCPU],
-				v1.ResourceMemory:           instanceType.Resources()[v1.ResourceMemory],
-				v1.ResourceEphemeralStorage: instanceType.Resources()[v1.ResourceEphemeralStorage],
-				v1alpha1.ResourceNVIDIAGPU:  instanceType.Resources()[v1alpha1.ResourceNVIDIAGPU],
-				v1alpha1.ResourceAMDGPU:     instanceType.Resources()[v1alpha1.ResourceAMDGPU],
-				v1alpha1.ResourceAWSNeuron:  instanceType.Resources()[v1alpha1.ResourceAWSNeuron],
-			} {
-				if !quantity.IsZero() {
-					resources[resourceName] = quantity
-				}
-			}
-
-			return &v1.Node{
+			n := &v1.Node{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: nodeName,
 					Labels: map[string]string{
 						v1.LabelTopologyZone:       aws.StringValue(instance.Placement.AvailabilityZone),
 						v1.LabelInstanceTypeStable: aws.StringValue(instance.InstanceType),
+						v1.LabelArchStable:         v1alpha1.AWSToKubeArchitectures[aws.StringValue(instance.Architecture)],
 						v1alpha5.LabelCapacityType: getCapacityType(instance),
 					},
 				},
 				Spec: v1.NodeSpec{
 					ProviderID: fmt.Sprintf("aws:///%s/%s", aws.StringValue(instance.Placement.AvailabilityZone), aws.StringValue(instance.InstanceId)),
 				},
-				Status: v1.NodeStatus{
-					Allocatable: resources,
-					Capacity:    resources,
-					NodeInfo: v1.NodeSystemInfo{
-						Architecture:    v1alpha1.AWSToKubeArchitectures[aws.StringValue(instance.Architecture)],
-						OSImage:         aws.StringValue(instance.ImageId),
-						OperatingSystem: v1alpha5.OperatingSystemLinux,
-					},
-				},
 			}
+
+			n.Labels[v1.LabelOSStable] = "linux"
+			if amiFamily != nil {
+				switch *amiFamily {
```

Comment:
If you use
`aws.StringValue(amiFamily)`

This will let you drop the extra nil check and layer of indentation/complexity.